### ISSUE TITLE
Function Groups vs Classes

### ISSUE THREAD

**Issue by HrFlorianHoffmann** [SAP/styleguides#83]
Received this input from internally. Translation by me.

Related to the sub-section [Function Groups vs. Classes](https://github.com/SAP/styleguides/blob/master/clean-abap/sub-sections/FunctionGroupsVsClasses.md).

> No substitution.
> A call to a function module is technically always dynamic.
> Therefore the syntax check for the signature is also only done at runtime.
> This means it is always possible to put the function module name into a variable and call it dynamically.
> This is also what frameworks like BDT and BRF+ do.
> The sentence in the guide's section thus is wrong.

> Variable encapsulation.
> There is no real privacy for variables of function groups.
> They can even be modified from the outside.
> This statement thus is plainly wrong.
> For example:

```
FIELD-SYMBOLS <fs>TYPE any.
ASSIGN ('(<report_name>)gv_global_variable') TO <fs>
```

> Method encapsulation.
> This statement too is simply and completely wrong.
> Form routines can always be called from the outside.
> For example:

```
PERFORM set_buffer_true IN PROGRAM <some_program>.
```

**Comment by schmelto** [comment]
think this issue can be closed?

**Comment by bjoern-jueliger-sap** [comment]
Fixed with #87.
